### PR TITLE
Avoid creating a separate visualization discretization when avoidable

### PR DESCRIPTION
**Pull request by inducer**
Towards https://github.com/inducer/grudge/issues/65.

cc @nchristensen

@alexfikl, what do you think? (I'd request a review, but somehow I can't?)

### CODE REVIEW

File: meshmode/discretization/visualization.py [hunk 1]
```diff
@@ -1067,22 +1067,35 @@ def make_visualizer(actx, discr, vis_order,
...
     from meshmode.discretization.poly_element import OrderAndTypeBasedGroupFactory
 
-    if force_equidistant:
-        from meshmode.discretization.poly_element import (
+    vis_discr = None
+    if element_shrink_factor is None and not force_equidistant:
+        if vis_order is None:
+            vis_discr = discr
+        else:
+            if all(grp.order == vis_order for grp in discr.groups):
```
Comment (alexfikl) [inline meshmode/discretization/visualization.py:1075]: Does this warrant a warning? 

The visualizer can also change the element group (e.g. from `InterpolatoryQuadrature` to `PolynomialWarpAbdBlend`), so in pytential it makes sense to pass the same order.

**Review by alexfikl — APPROVED** [review]
Looks good to me! 

Big +1 on the `IdentityDiscretizationConnection`. I've been using an empty `ChainedDiscretizationConnection` for that, which didn't feel right.

**Comment by alexfikl** [comment]
For the pytential failure, it should just need skipping non-direct connections somewhere around [here](https://github.com/inducer/meshmode/blob/1a61acc202da40fe9582a42bd07991f151d7d1da/meshmode/discretization/connection/chained.py#L212-L214)

File: meshmode/discretization/visualization.py (hunk 1 above)
Comment (inducer) [reply meshmode/discretization/visualization.py]: Fair point. How's f3ff137c990ab6198e299a500f9c66832ca4bf0d?

Comment (alexfikl) [reply meshmode/discretization/visualization.py]: I can't seem to resolve this, so :+1:

**Comment by inducer** [comment]
Thanks for taking a look!

**Pull request merged**